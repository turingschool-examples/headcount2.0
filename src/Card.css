.card {
  display: grid;
  justify-content: center;
  align-content: center;
  border-radius: 5px;
  border: 1px solid black;
}

.card li {
  list-style: none;
}

.green {
  color: green;
}

.red {
  color: red;
}