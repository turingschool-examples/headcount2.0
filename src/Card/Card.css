.card {
    box-shadow: 5px 5px 8px;
    width: 185px;
    height: 233px;
    border: 1px #000000 solid;
    margin: 7px;
}

ul {
  list-style-type: none;
}

li {
  font-size: 12px;
}

h3 {
  text-align: center;
  font-size: 15px;
  padding: 0 7px;
}