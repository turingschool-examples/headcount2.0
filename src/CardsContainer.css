.cards-container {
  display: grid;
  grid-gap: 5px;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
}