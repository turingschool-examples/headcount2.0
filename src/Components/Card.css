.card {
  background-color: #375053;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

.selected {
  background-color: #5e7377;
  border: solid 4px #f7b488;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 6em;
}

.stat {
  padding: 0 0 .5em 0;
}

.less-than {
  color: #f7b488;
}