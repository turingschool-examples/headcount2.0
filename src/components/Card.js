import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ location, data, handleCompare, compare, active }) => {
  const classname = active === true ? 'card active' : 'card'
  return (
    <div className={classname} key={location} onClick={() => handleCompare(location)}>
      <h1 className='cardTitle'>
        {location}
      </h1>
      {Object.keys(data).map((key, index) => {


        return (
          <div className='cardData' key={index}>
            <li className={
              data[key] <= 0.5 ? 'negative' : 'positive'
            }> {key}: {data[key]} </li>
          </div>
        )
      })}
    </div>
  );
}

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.object,
  handleCompare: PropTypes.func,
}

export default Card;
