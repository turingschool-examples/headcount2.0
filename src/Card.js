import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({location, stats}) => {
  const statsKeys = Object.keys(stats);
  const resultsLi = statsKeys.map((years, i) => 
    <li 
      key={Date.now() + i}
      className={stats[years] < 0.5 ? "red" : "green"}
    >
    { years } : { stats[years] }
    </li>)

  return (
    <div className="card">
      <h3>{location}</h3>
      <ul>
        {resultsLi}
      </ul>
    </div>
  )
}

Card.propTypes = {
  location: PropTypes.string,
  stats: PropTypes.object
}

export default Card;