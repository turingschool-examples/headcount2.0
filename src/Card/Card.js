import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Card.css';

class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clicked: this.props.clickedState || false
    };
  }

  handleCardClick = () => {
    
    if (this.props.comparedCards.length === 2  && 
      !this.props.comparedCards.find(district => {
        return district.district.location === this.props.district.location;
      })) {
      return;
    }
    this.setState({ clicked: !this.state.clicked });
    this.props.updateCompareState(this.state.clicked, this.props);
  }

  render() {
    let districtKeys = Object.keys(this.props.district.stats);
    const listItems = districtKeys.map(( stat, index ) => {
      return (
        <li 
          key={ index }
          className={ this.props.district.stats[stat] > 0.5 ? 'greater-than' : 'less-than'}
        > { `${stat}: ${this.props.district.stats[stat]}` } </li>
      );
    });
  
    return (
      <div 
        className={this.state.clicked ? 'card clicked' : 'card'} 
        onClick={ this.handleCardClick }
      > 
        <h3>
          { this.props.district.location }
        </h3>
        <ul>
          { listItems }
        </ul>
      </div>
    );
  }
}

Card.propTypes = {
  district: PropTypes.string.isRequired,
  comparedCards: PropTypes.arrayOf(PropTypes.object),
  updateCompareState: PropTypes.func.isRequired,
  clickedState: PropTypes.bool
};



export default Card;