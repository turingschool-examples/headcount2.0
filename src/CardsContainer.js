import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './CardsContainer.css';


const CardsContainer = ({schoolData}) => {

  const listLocations = Object.keys(schoolData);
  const locationCards = listLocations.map((location, i) =>
    <Card 
      {...schoolData[location] } key={Date.now() + i}
    />)
  return (
    <div className = "cards-container">
      { locationCards }
    </div>
  )
}

CardsContainer.propTypes = {
  schoolData: PropTypes.array
}


export default CardsContainer;