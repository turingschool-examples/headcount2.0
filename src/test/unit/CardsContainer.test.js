import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CardsContainer from '../../../src/CardsContainer';
import Card from '../../../src/Card';

describe('CardsContainer', () => {
  it('should render 181 cards', () => {
    const wrapper = shallow(<CardsContainer schoolData={[]} />);    
    
    expect(wrapper).toMatchSnapshot();
  })

  it('should render all the the cards', () => {
    const mockData = [{title: 'hey', body: 'yall'}, {title: 'war', body: 'eagle'}]
    const wrapper = shallow(<CardsContainer schoolData={mockData} />);

    expect(wrapper.find(Card).length).toEqual(2);
  })

})
