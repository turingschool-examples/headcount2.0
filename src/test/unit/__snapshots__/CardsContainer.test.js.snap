// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer should render 181 cards 1`] = `
<div
  className="cards-container"
/>
`;
